File: fapacapp/src/app/model/PesquisadorDto.ts
import { TipoPesquisador } from './TipoPesquisador';  

export interface PesquisadorDto {
    idPesquisador: number;
    nomeCompletoPesquisador: string;
    fotoPesquisador: string;
    emailPesquisador: string;
    sexoPesquisador: string;
    dataNascimentoPesquisador: string;  
    corRaca: string;
    nomeMaePesquisador: string;
    nomePaiPesquisador: string;
    curriculoLattesPesquisador: string;
    nivelAcademicoPesquisador: string;
    areaConhecimentoumPesquisador: string;
    areaConhecimentodoisPesquisador: string;
    areaConhecimentotresPesquisador: string;
    cpfPesquisador: string;
    cpfFilePesquisador: string;
    senhaPesquisador: string;
    rgPesquisador: string;
    rgFilePesquisador: string;
    orgaoEmissorPesquisador: string;
    ufPesquisador: string;
    dataEmissaoPesquisador: string;
    passaportePesquisador: string;
    passaporteFilePesquisador: string;
    telefonePesquisador: string;
    tipoPesquisador: TipoPesquisador;

    // Endereço Residencial
    idEnderecoPesquisador: number;
    tipoEnderecoPesquisador: string;
    cepEnderecoPesquisador: string;
    logradouroEnderecoPesquisador: string;
    numeroEnderecoPesquisador: string;
    bairroEnderecoPesquisador: string;
    paisEnderecoPesquisador: string;
    estadoEnderecoPesquisador: string;
    municipioEnderecoPesquisador: string;
    telefoneEnderecoPesquisador: string;

    // Vínculo Institucional
    idvinculoInstitucional: number;
    vinculoInstitucional: string;
    vinculoEmpregaticio: boolean;
    tempoServico: string;
    funcaoCargo: string;
    tempoFuncao: string;

    // Endereço Profissional
    idunidade: number;
    cepEnderecoUnidade: string;
    logradouroEnderecoUnidade: string;
    numeroEnderecoUnidade: string;
    bairroEnderecoUnidade: string;
    estadoEnderecoUnidade: string;
    paisEnderecoUnidade: string;
}

File: fapacapi/src/main/java/fapacapi/controller/dto/PesquisadorDto.java
package fapacapi.controller.dto;

import java.time.LocalDate;
import java.util.List;

import fapacapi.model.PesquisadorConhecimento;
import fapacapi.model.VinculoInstitucional;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record PesquisadorDto(
        @NotNull Long idPesquisador,
        @NotBlank String nomeCompletoPesquisador,
        @NotBlank String fotoPesquisador,
        @NotBlank String emailPesquisador,
        @NotBlank String sexoPesquisador,
        @NotNull LocalDate dataNascimentoPesquisador,
        @NotBlank String corRaca,
        @NotBlank String nomeMaePesquisador,
        @NotBlank String nomePaiPesquisador,
        @NotBlank String curriculoLattesPesquisador,
        @NotBlank String nivelAcademicoPesquisador,
        @NotBlank String areaConhecimentoumPesquisador,
        @NotBlank String areaConhecimentodoisPesquisador,
        @NotBlank String areaConhecimentotresPesquisador,
        @NotBlank String cpfPesquisador,
        @NotBlank String cpfFilePesquisador,
        @NotBlank String senhaPesquisador,
        @NotBlank String rgPesquisador,
        @NotBlank String rgFilePesquisador,
        @NotBlank String orgaoEmissorPesquisador,
        @NotBlank String ufPesquisador,
        @NotBlank String dataEmissaoPesquisador,
        @NotBlank String passaportePesquisador,
        @NotBlank String passaporteFilePesquisador,
        @NotBlank String telefonePesquisador,
        @NotNull Long idTipoPesquisador,

        // Endereço Residencial
        @NotNull Long idEnderecoResidencial, 
        @NotBlank String tipoEnderecoResidencial,
        @NotBlank String cepEnderecoResidencial,
        @NotBlank String logradouroEnderecoResidencial,
        @NotBlank String numeroEnderecoResidencial,
        @NotBlank String bairroEnderecoResidencial,
        @NotBlank String paisEnderecoResidencial,
        @NotBlank String estadoEnderecoResidencial,
        @NotBlank String municipioEnderecoPesquisador,
        @NotBlank String telefoneEnderecoPesquisador,
        
        // vinculo institucional
        // @NotNull Long idvinculoInstitucional,
        // @NotBlank String vinculoInstitucional,
        // @NotNull Boolean vinculoEmpregaticio,
        // @NotBlank String tempoServico,
        // @NotBlank String funcaoCargo,
        // @NotBlank String tempoFuncao,
        
        //ENDEREÇO PROFISSIONAL
        // @NotNull Long idUnidade,
        // @NotBlank String cepEnderecoUnidade,
        // @NotBlank String logradouroEnderecoUnidade,
        // @NotBlank String numeroEnderecoUnidade,
        // @NotBlank String bairroEnderecoUnidade,
        // @NotBlank String estadoEnderecoUnidade,
        // @NotBlank String paisEnderecoUnidade

        Long idEnderecoProfissional, 
        String tipoEnderecoProfissional,
        String cepEnderecoProfissional,
        String logradouroEnderecoProfissional,
        String numeroEnderecoProfissional,
        String bairroEnderecoProfissional,
        String paisEnderecoProfissional,
        String estadoEnderecoProfissional,
        String municipioEnderecoProfissional,
        String telefoneEnderecoProfissional,

        // Vinculo Institucional
        // Lista de Vínculos Institucionais
        @NotNull List<VinculoInstitucional> vinculosInstitucionais,

        // PesquisadorConhecimento
        @NotNull List<PesquisadorConhecimento> pesquisadorConhecimento

) {

}
